environment_setup/install_cubemx.sh

```diff
@@ -39,12 +39,22 @@ else
         </AutomatedInstallation>"
     
     echo -n $AUTO_INSTALL_XML > auto-install.xml
+
+    # create script to cd and run STM32CubeMX
+    STM32_SCRIPT="cd /opt/STM32CubeMX_$CUBE_VERSION && ./STM32CubeMX && cd -"
```

Comment:
Consider using pushd and popd

Reply:
:+1: for this, much less error-prone

Reply:
not opposed to changing this, just wondering what makes pushd/popd less error prone?

Reply:
Perhaps "much" was a bit of a strong word, but IMO writing:
```
pushd /tmp/stm32cubemx

do something 
do something else

popd
```
provides a much clearer indication that you're moving into a directory, then back to the directory you started in. Although TIL what `cd -` does, that's basically the same. Don't feel super strongly here either way, leaving this up to you / Jon.

Reply:
yeah its definitely more explicit, fixed!